intelliserver/api/ocr/index.js

```diff
@@ -0,0 +1,314 @@
+ *                 data:
+ *                   type: object
+ *                   properties:
+ *                     text:
+ *                       type: array
+ *                       items: 
+ *                          type: string
+ *                       description: The extracted text from the image.
+ *       400:
+ *         description: Invalid request or image format.
+ *       500:
+ *         description: Internal server error.
+ *     parameters:
+ *       - in: header
+ *         name: X-aws-access-Key
+ *         schema:
+ *           type: string
+ *         required: false
+ *         description: Optional AWS access key for this specific request.
+ *       - in: header
+ *         name: X-aws-secret-Key
+ *         schema:
+ *           type: string
+ *         required: false
+ *         description: Optional AWS secret key for this specific request.
+ *       - in: header
+ *         name: X-aws-region
+ *         schema:
+ *           type: string
+ *         required: false
+ *         description: Optional AWS region for this specific request.
+ */
+
+/**
+ * @swagger
+ * /ocr/google:
+ *   post:
+ *     summary: Perform OCR on an image using Google Cloud Vision API.
+ *     tags:
+ *       - OCR
+ *     requestBody:
+ *       required: true
+ *       content:
+ *         application/json:
+ *           schema:
+ *             type: object
+ *             properties:
+ *               imageUrl:
+ *                 type: string
+ *                 description: The URL of the image to perform OCR on.
+ *     responses:
+ *       200:
+ *         description: OCR results.
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 status:
+ *                   type: string
+ *                   description: The status of the OCR operation (e.g., "OK").
+ *                 data:
+ *                   type: object
+ *                   properties:
+ *                     text:
+ *                       type: string
+ *                       description: The extracted text from the image.
+ *       400:
+ *         description: Invalid request or image URL.
+ *       500:
+ *         description: Internal server error.
+ *     parameters:
+ *       - in: header
+ *         name: X-aws-access-Key
+ *         schema:
+ *           type: string
+ *         required: false
+ *         description: Optional AWS access key for this specific request.
+ *       - in: header
+ *         name: X-aws-secret-Key
+ *         schema:
+ *           type: string
+ *         required: false
+ *         description: Optional AWS secret key for this specific request.
+ *       - in: header
+ *         name: X-aws-region
+ *         schema:
+ *           type: string
+ *         required: false
+ *         description: Optional AWS region for this specific request.
+ */
+
+router.post('/google', getImageFromUrlOrFile, async (req, res) => {
```

Comment:
did you test google ocr ?

Reply:
Not yet, I've created Vision AI app but it needs one more day to activate the billing account for that